ex7.3_post_office2.py

```diff
@@ -0,0 +1,124 @@
+            'body': message_body,
+            'sender': sender,
+        }
+        if urgent:
+            user_box.insert(0, message_details)
+        else:
+            user_box.append(message_details)
+        return self.message_id
+
+    def read_inbox(self, username, N=10):
+        """
+        An function that gets the name of the user and number of the message request.
+        the function returns the first N messages in the mailbox.
+        :param username: name of the user
+        :param N: number of the message request
+        :return: the first N messages in the mailbox. (if not exist return all messages in the mailbox)
+        """
+        return_message = []
+        count_message = 0
+        if not N == 0:
+            if N < len(self.boxes[username]):
+                for message in range(len(self.boxes[username])):
+                    if self.boxes[username][message].message_details['status'] == 'not read':
+                        self.boxes[username][message].message_details['status'] == 'read'
+                        count_message = count_message + 1
+                        return_message.append(self.boxes[username][message])
+                    if count_message == N:
+                        break
+
+        for message in range(len(self.boxes[username])):
+            if self.boxes[username][message].message_details['status'] == 'not read':
+                self.boxes[username][message].message_details['status'] == 'read'
+                return_message.append(self.boxes[username][message])
+        return return_message
+
+    def search_inbox(self, username, sentence):
+        """
+        An function that gets the name of the user and string .
+        the function return list of all the messages that contains the string in there headline or font.
+        :param username:  the name of the user
+        :param sentence:  string
+        :return: list of all the messages that contains the string in there headline or font.
+        """
+        return_message = []
+        for message in self.boxes[username]:
+            if sentence in message.message_details['body'] or sentence in message.message_details['sender']:
+                return_message.append(message)
+
+        return return_message
+
+
+class Message:
+    """
+    class message - print and help with the details
+    """
+
+
+def __init__(self, message_id, message_body, message_sender, message_status="not read"):
+    """
+    the init function that dave the information of the message that necessary
+    :param self: the self
+    :param message_id: the message id
+    :param message_body: body of the message - the text
+    :param message_sender: the message send
+    :param message_status: the message status
+    :return:
+    """
+    self.id = message_id
+    self.body = message_body
+    self.sender = message_sender
+    self.status = message_status
+
+
+def __str__(self):
+    """
+    function that prints the message.
+    """
```

Comment:
Read this link and then properly implement a _ _ str _ _ function
https://www.pythontutorial.net/python-oop/python-__str__/